test/test_type_promotion.py

```diff
@@ -430,6 +431,123 @@ def test_transpose(self, device):
         a = torch.tensor([[True, True], [False, True]], device=device)
         self.assertEqual(a.t() == 0, a.t() == False)  # noqa: E712
 
+    def _test_sparse_op_input_tensors(self, device, dtype, coalesced, zeros=True):
+        t = self._get_test_tensor(device, dtype, not zeros)
+        if zeros and dtype != torch.bool:
+            # ensure sparsity. Bool should already have sufficient sparsity.
+            mask = self._get_test_tensor(device, torch.bool)
+            t = t * mask
+        if coalesced:
+            s = t.to_sparse()
+        else:
+            s = t.to_sparse()
+            indices = torch.cat((s.indices(), s.indices()), 1)
+            values = torch.cat((s.values(), s.values()), 0)
```

Comment:
nit: Might be easier to read if we used `s = s + s` (because sparse addition always appends the values and indices). Since we have the assertions below anyways to check that things are correct, if we ever change the implementation then we'll have an alert

Reply:
I'd find that harder to read since it requires thinking about the implementation of add. cat is a bit more explicit about what's happening.

Reply:
The reason why I find `s = s + s` easier to read is because I don't have to reason about if this code is cat-ing the correct things together. A comment on top of it "the implementation of add is to just cat the values and indices" clears up that point, but this is just a nit.